src/Microsoft.Sbom.Api/Output/Telemetry/TelemetryRecorder.cs

```diff
@@ -194,6 +197,22 @@ public void RecordException(Exception exception)
         this.exceptions.Add(exception);
     }
 
+    public void RecordTotalNumberOfPackages(int packageCount)
+    {
+        this.totalNumberOfPackages += packageCount;
```

Comment:
It seems like this `IRecorder` gets registered as a singleton which might be accessed by multiple threads, but this operation is not thread-safe. Our counter may therefore be incorrect in certain race conditions.

Reply:
Ahh, you're only doing a console App for now (got confused by the name `Microsoft.Sbom.Api`. This would be safe for now then, though we might put a warning on the class/method that it's not thread-safe if it ever might be a real API (I see a bunch of Dictionaries in here which would also be issues).

Reply:
This library also getting shipped as nuget package and sbom generation/validation utilizes multithreading. I believe this might be a valid concern that worth investigating

Reply:
Made a change that I believe would help with the addition part. Are the dictionaries you are referring to part of my changes or were they there already?

Reply:
@sebasgomez238 and I talked offline, about the += stuff, but WRT the `IDictionary`, `IList` and `HashSet` stuff that's pre-existing (other than the HashSet that was added).